src/gtk2-1-fixes.patch

```diff
@@ -567,3 +567,161 @@ index 1111111..2222222 100644
++  extern void _gtk_theme_wimp_exit (void);
++  _gtk_theme_wimp_exit();
++#else
+   engine->exit();
+ 
+   g_module_close (engine->library);
+@@ -119,6 +128,7 @@ gtk_theme_engine_unload (GTypeModule *module)
+   engine->init = NULL;
+   engine->exit = NULL;
+   engine->create_rc_style = NULL;
++#endif
+ }
+ 
+ static void
+@@ -189,8 +199,32 @@ gtk_theme_engine_create_rc_style (GtkThemeEngine *engine)
+ {
+   g_return_val_if_fail (engine != NULL, NULL);
+ 
++#ifdef G_OS_WIN32
++  extern GtkRcStyle * _gtk_theme_wimp_create_rc_style (void);
++  return _gtk_theme_wimp_create_rc_style ();
++#else
+   return engine->create_rc_style ();
++#endif
+ }
+ 
+ #define __GTK_THEMES_C__
+ #include "gtkaliasdef.c"
++
++#ifdef G_OS_WIN32
++/* redefine deprected */
++#include "gtknotebook.h"
++GType gtk_clist_get_type (void) G_GNUC_CONST;
++#define GTK_TYPE_CLIST            (gtk_clist_get_type ())
++#define GTK_IS_CLIST(obj)         (G_TYPE_CHECK_INSTANCE_TYPE ((obj), GTK_TYPE_CLIST))
++#define GTK_WIDGET_VISIBLE(wid)		  ((GTK_WIDGET_FLAGS (wid) & GTK_VISIBLE) != 0)
++void gtk_notebook_query_tab_label_packing (GtkNotebook *notebook,
++					   GtkWidget   *child,
++					   gboolean    *expand,
++					   gboolean    *fill,
++					   GtkPackType *pack_type);
++
++#include "../modules/engines/ms-windows/msw_rc_style.c"
++#include "../modules/engines/ms-windows/msw_style.c"
++#include "../modules/engines/ms-windows/msw_theme_main.c"
++#include "../modules/engines/ms-windows/xp_theme.c"
++#endif
+\ No newline at end of file
+diff --git a/modules/engines/ms-windows/msw_rc_style.c b/modules/engines/ms-windows/msw_rc_style.c
+index 1111111..2222222 100755
+--- a/modules/engines/ms-windows/msw_rc_style.c
++++ b/modules/engines/ms-windows/msw_rc_style.c
+@@ -27,7 +27,7 @@ static void msw_rc_style_init (MswRcStyle * style);
+ static void msw_rc_style_class_init (MswRcStyleClass * klass);
+ static GtkStyle *msw_rc_style_create_style (GtkRcStyle * rc_style);
+ 
+-static GtkRcStyleClass *parent_class;
++static GtkRcStyleClass *rcparent_class;
+ 
+ GType msw_type_rc_style = 0;
+ 
+@@ -62,7 +62,7 @@ msw_rc_style_class_init (MswRcStyleClass *klass)
+ {
+   GtkRcStyleClass *rc_style_class = GTK_RC_STYLE_CLASS (klass);
+ 
+-  parent_class = g_type_class_peek_parent (klass);
++  rcparent_class = g_type_class_peek_parent (klass);
+ 
+   rc_style_class->create_style = msw_rc_style_create_style;
+ }
+diff --git a/modules/engines/ms-windows/msw_theme_main.c b/modules/engines/ms-windows/msw_theme_main.c
+index 1111111..2222222 100755
+--- a/modules/engines/ms-windows/msw_theme_main.c
++++ b/modules/engines/ms-windows/msw_theme_main.c
+@@ -91,8 +91,13 @@ global_filter_func (void *xevent, GdkEvent *event, gpointer data)
+     }
+ }
+ 
+-G_MODULE_EXPORT void
+-theme_init (GTypeModule * module)
++#ifdef G_OS_WIN32
++#define MODULE_ENTRY(type, function) type _gtk_theme_wimp_ ## function
++#else
++#define MODULE_ENTRY(type, function) G_MODULE_EXPORT type theme_ ## function
++#endif
```

Comment:
Are you sure G_MODULE_EXPORT isn't needed for G_OS_WIN32? Have you tested static and shared builds?

Reply:
This was mostly based on an old patch that I migrated to a newer gtk, however as for 64bit shared builds, gtk2 never built for me in shared form period so even before this patch it still didn't exactly compile period in that form on any of my rigs. My test case showed it does work, but I have not extensively tested it outside of static builds.